File: algo/c++/sentence_order_and_word_reverse.cpp
#include <iostream>
using namespace std;
int main () {
    char sentence[] = "sentence order and word reverse";

    char *index = sentence;
    char *start = sentence;
    char *end = sentence;

    while (true) {
        start = index;
        for (; *index != ' ' && *index != '\0'; index++);
        end = index - 1;
        for (char tmp; start < end; start++, end--) {
            tmp = *start;
            *start = *end;
            *end = tmp;
        }
        if (*index++ == '\0') break;
    }

    cout << sentence << endl;

    return 0;
}

File: algo/javascript/sentence_order_and_word_reverse.js
var sentence = "sentence order and word reverse";
var result = '', tmp = '';
var index = 0;

while (true) {
    for (; sentence[index] !== ' ' && sentence[index] !== void 0; index++) {
        tmp = sentence[index] + tmp;
    }

    result += tmp + (sentence[index] || '');

    tmp = '';

    if (sentence[index++] === void 0) {
        break;
    }
}

console.log(result); // ecnetnes redro dna drow esrever
